include/calculate/symbols.hpp

```diff
@@ -109,14 +114,50 @@ namespace symbols {
     public:
         const String token;
         const Type type;
-        bool is(Type y) noexcept {return type == y;}
+        virtual bool is(Type y) noexcept = 0;
```

Comment:
This trick was necessary to conserve the `Symbol` class as an **Abstract Data Type**, since it lost its single pure virtual method, `evaluate`.